libs/wire-subsystems/test/unit/Wire/UserSubsystem/InterpreterSpec.hs

```diff
@@ -878,3 +883,36 @@ spec = describe "UserSubsystem.Interpreter" do
               === ( ChangeEmailResponseNeedsActivation,
                     [user {emailUnvalidated = Just updatedEmail}]
                   )
+
+  describe "GuardRegisterUser" $
+    prop "throws the appropriate errors" $
+      \(domreg :: DomainRegistration) (preEmail :: EmailAddress) ->
+        let email :: EmailAddress
+            email = unsafeEmailAddress l (cs d)
+              where
+                l :: ByteString = localPart preEmail
+                d :: Text = domainText domreg.domain
+
+            interp ::
+              Sem
+                '[ DRS.DomainRegistrationStore,
+                   State [DRS.StoredDomainRegistration],
+                   TinyLog,
+                   Error UserSubsystemError
+                 ]
+                () ->
+              Either UserSubsystemError ()
+            interp = run . runError . noopLogger . evalState [] . inMemoryDomainRegistrationStoreInterpreter
```

Comment:
Should this interpreter be defined outside the test implementation, maybe?

Reply:
not sure.  i think it's too special-purpose to be moved to MockInterpreters, and it's not very long or complex.  i'd prefer to keep it here?

Reply:
That's ok, I meant, define it in the same module but outside the test.